frontends/web/src/components/language/language.tsx

```diff
@@ -65,127 +55,122 @@ const defaultLanguages = [
-              {
-                languages.map((language, i) => {
-                  const selected = selectedIndex === i;
-                  return (
-                    <button
-                      key={language.code}
-                      className={[style.language, selected ? style.selected : ''].join(' ')}
-                      onClick={() => this.changeLanguage(language.code, i)}
-                      data-index={i}
-                      data-code={language.code}>
-                      {language.display}
-                      {
-                        selected && (
-                          <svg
-                            className={style.checked}
-                            xmlns="http://www.w3.org/2000/svg"
-                            width="24"
-                            height="24"
-                            viewBox="0 0 24 24"
-                            fill="none"
-                            stroke="currentColor"
-                            strokeWidth="2"
-                            strokeLinecap="round"
-                            strokeLinejoin="round">
-                            <polyline points="20 6 9 17 4 12" />
-                          </svg>
-                        )
-                      }
-                    </button>
-                  );
-                })
-              }
-            </Dialog>
-          )
-        }
-      </div>
-    );
+
+  if (allLanguages.length === 1) {
+    return null;
   }
-}
 
-const TranslatedLanguageSwitcher = withTranslation()(LanguageSwitch);
+  return (
+
+    <div>
+      <button
+        title="Select Language"
+        className={style.link}
+        onClick={() => setActiveDialog(true)}>
+        <svg
+          xmlns="http://www.w3.org/2000/svg"
+          width="24"
+          height="24"
+          viewBox="0 0 24 24"
+          fill="none"
+          stroke="currentColor"
+          strokeWidth="2"
+          strokeLinecap="round"
+          strokeLinejoin="round">
+          <circle cx="12" cy="12" r="10" />
+          <line x1="2" y1="12" x2="22" y2="12" />
+          <path d="M12 2a15.3 15.3 0 0 1 4 10 15.3 15.3 0 0 1-4 10 15.3 15.3 0 0 1-4-10 15.3 15.3 0 0 1 4-10z" />
+        </svg>
+        {allLanguages[selectedIndex].code === 'en' ? 'Other languages' : 'English'}
+      </button>
+      {
+        activeDialog && (
+          <Dialog small slim title={t('language.title')} onClose={abort}>
+            {
+              allLanguages.map((language, i) => {
+                const selected = selectedIndex === i;
+                return (
+                  <button
+                    key={language.code}
+                    className={[style.language, selected ? style.selected : ''].join(' ')}
+                    onClick={() => changeLanguage(language.code, i)}
+                    data-index={i}
+                    data-testid={`language-selection-${language.code}`}
+                    data-code={language.code}>
```

Comment:
Looks like `data-code` and `data-index` was used once for changeLanguage but this changed in https://github.com/digitalbitbox/bitbox-wallet-app/commit/29a0764ea4a5d4f412ec928055439a083c47d99d#diff-42ba918f486c455a1e321c302b79b1e1931ce8c8e615412739e7e4ac92635666L138-R105

and now they are not used anymore. Keep only `data-testid` (https://testing-library.com/docs/queries/bytestid/) and remove the unused data attributes.